app/controllers/admin/bulk_line_items_controller.rb

```diff
@@ -73,5 +78,24 @@ def authorize_update!
     def order
       @line_item.order
     end
+
+    def order_permissions
+      ::Permissions::Order.new(spree_current_user)
+    end
+
+    def using_pagination?
+      params[:per_page]
+    end
+
+    def pagination_data
+      return unless using_pagination?
+
+      {
+        results: @line_items.total_count,
+        pages: @line_items.num_pages,
+        page: params[:page].to_i,
```

Comment:
this should be params[:page] || 1
probably better to have a method page and use it here and above
def page
    params[:page] || 1
end

Reply:
I've added this into the original commit. All the other Angular refactoring bits are in a new commit.